src/components/AskForPushNotification.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { t } from 'ttag';
+import ActionModal from './ActionModal';
+import { pushDismissOptInQuestion } from '../actions';
+
+const mapStateToProps = (state) => ({
+  optInDismissed: state.pushNotification.optInQuestionDismissed,
+});
+
+const mapDispatchToProps = (dispatch) => ({
+  dismissOptInQuestion: () => dispatch(pushDismissOptInQuestion()),
+});
+
+const onEnablePushNotifications = (props) => {
+  props.navigation.navigate('PushNotification');
+  props.dismissOptInQuestion();
+};
+
+function AskForPushNotification(props) {
+  if (props.optInDismissed) return null;
+  return (
+    <ActionModal
+      title={t`Do you want to enable push notifications for this wallet?`}
+      message={t`You can always change this later in the settings menu.`}
+      button={t`Yes, enable`}
+      onAction={() => onEnablePushNotifications(props)}
+      onDismiss={() => props.dismissOptInQuestion()}
+    />
+  );
+}
```

Comment:
Since we are already using function components, we can use [react hooks](https://reactjs.org/docs/hooks-intro.html) here:

```javascript
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { pushDismissOptInQuestion } from '../actions';

(...)

function AskForPushNotification ({ navigation }) {
  const dispatch = useDispatch();
  const optInDismissed = useSelector(({ pushNotification }) => pushNotification.optInQuestionDismissed);

  if (!optInDismissed) {
    return null;
  }

  const onAction = () => {
    navigation.navigate('PushNotification');
    dispatch(pushDismissOptInQuestion());
  };

  const onDismiss = () => dispatch(pushDismissOptInQuestion());

  return (
    <ActionModal
      title={t`Do you want to enable push notifications for this wallet?`}
      message={t`You can always change this later in the settings menu.`}
      button={t`Yes, enable`}
      onAction={() => onAction()}
      onDismiss={() => onDismiss()}
    />
  );
}
```

Reply:
We could also use `useNavigation` to avoid needing to pass the navigation prop down the component tree, but it's not available in the current version of react-navigation

Reply:
I didn't know about the `useNavigation`, and I did this PR before the Settings refactoring. I can refactor this one as well.

Reply:
Yep. Moving without `useNavigation`.